model/host/hostutil.go

```diff
@@ -498,20 +500,75 @@ SeServiceLogonRight = $($currentSetting)
 `}, "\r\n"), nil
 }
 
+const passwordCharset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
+
+func generatePassword(length int) string {
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = passwordCharset[rand.Int()%len(passwordCharset)]
+	}
+	return string(b)
+}
+
 // CreateServicePassword creates the password for the host's service user.
 func (h *Host) CreateServicePassword() error {
-	password := util.RandomString()
+	var password string
+	var valid bool
+	for i := 0; i < 1000; i++ {
+		password = generatePassword(12)
+		if strings.ToLower(password) == password {
+			continue
+		}
+		if valid = ValidateRDPPassword(password); valid {
+			break
+		}
+	}
+	if !valid {
+		return errors.New("could not generate valid service password")
+	}
+
 	err := UpdateOne(
 		bson.M{IdKey: h.Id},
 		bson.M{"$set": bson.M{ServicePasswordKey: password}},
 	)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "could not update service password")
 	}
 	h.ServicePassword = password
 	return nil
 }
 
+// each regex matches one of the 5 categories listed here:
+// https://technet.microsoft.com/en-us/library/cc786468(v=ws.10).aspx
+var passwordRegexps = []*regexp.Regexp{
```

Comment:
Can you also delete the duplicate code from the cloud package and have it call this?

Reply:
Removed.